docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,552 @@
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      env:
+        - name: CPU_LIMIT
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.containers[?(@.name=="container-name")].resources.limits.cpu
+```
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kubernetes-downwardapi-volume-example
+spec:
+  containers:
+    - name: client-container
+      image: gcr.io/google_containers/busybox
+      command: ["sh", "-c", "while true; do if [[ -e /etc/labels ]]; then cat /etc/labels; fi; if [[ -e /etc/annotations ]]; then cat /etc/annotations; fi;sleep 5; done"]
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      volumeMounts:
+        - name: podinfo
+          mountPath: /etc
+          readOnly: false
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "cpu_limit"
+            fieldRef:
+              fieldPath: spec.containers[?(@.name=="container-name")].resources.limits.cpu
+```
+
+#### Validations
+
+For APIs with full json path selectors, verify that selectors are
+valid relative to pod spec.
+
+
+### API with partial JSONpath selectors
+
+Partial json path selectors specify paths to resources limits and requests
+relative to the container spec. These will be implemented by introducing a
+`ContainerFieldSelector` (json: `containerSpecFieldRef`) to extend the current
+implementation for `type DownwardAPIVolumeFile struct` and `type EnvVarSource struct`.
+
+```
+// ContainerFieldSelector selects an APIVersioned field of an object.
+type ContainerFieldSelector struct {
+        APIVersion string `json:"apiVersion"`
+        // Container name
+        Name string `json:"containerName,omitempty"`
+        // Required: Path of the field to select in the specified API version
+        FieldPath string `json:"fieldPath"`
+}
+
+// Represents a single file containing information from the downward API
+type DownwardAPIVolumeFile struct {
+     // Required: Path is  the relative path name of the file to be created.
+     Path string `json:"path"`
+     // Selects a field of the pod: only annotations, labels, name and
+     // namespace are supported.
+     FieldRef *ObjectFieldSelector `json:"fieldRef, omitempty"`
+     // Selects a field of the container: only resources limits and requests
+     // (resources.limits.cpu, resources.limits.memory, resources.requests.cpu,
+     // resources.requests.memory) are currently supported.
+     ContainerFieldRef *ContainerFieldSelector `json:"containerSpecFieldRef,omitempty"`
+}
+
+// EnvVarSource represents a source for the value of an EnvVar.
+// Only one of its fields may be set.
+type EnvVarSource struct {
+   // Required: Selects a field of the container: only resources limits and
+   // requests (cpu, memory) are supported.
```

Comment:
Please document the exact expected syntax in the comment.

Also, this field can't be required.